COMETwebapp/Components/Viewer/BabylonCanvasBase.cs

```diff
@@ -122,31 +113,47 @@ protected async override Task OnAfterRenderAsync(bool firstRender)
         /// </summary>
         private void InitializeElements()
         {
-            if(this.SessionAnchor != null)
+            var iteration = this.SessionAnchor?.OpenIteration;
+
+            var elementUsages = this.SessionAnchor?.OpenIteration?.Element.SelectMany(ed => ed.ContainedElement).OrderBy(x=>x.Name).ToList();
+            
+            if(elementUsages is not null)
             {
-                var iteration = this.SessionAnchor?.OpenIteration;
+                foreach (var elementUsage in elementUsages)
+                {
+                    var parameter = elementUsage.ElementDefinition.Parameter.FirstOrDefault(x => x.ParameterType.Name == "Shape kind"
+                                          && x.ParameterType is EnumerationParameterType
+                                          || x.ParameterType is TextParameterType);
 
-                var elementUsages = iteration?.Element.SelectMany(x => x.ContainedElement).ToList();
+                    Primitive basicShape = null;
+                    if (parameter is not null)
+                    {
+                        string? shapekind = parameter?.ExtractActualValues(1).First();
+                        switch (shapekind?.ToLowerInvariant())
+                        {
+                            case "box": basicShape = new Cube(1, 1, 1); break;
+                        }
+                    }
+                    parameter = elementUsage.ElementDefinition.Parameter.FirstOrDefault(x => x.ParameterType.Name == "Position"
+                                                          && x.ParameterType is CompoundParameterType);
 
-                if (elementUsages != null)
-                {
-                    foreach (var elementUsage in elementUsages)
+                    string[]? translations = parameter?.ExtractActualValues(3);
+                    if (basicShape is PositionablePrimitives positionablePrim && translations is not null)
```

Comment:
These kind of things should be refacottred to be in own methods that can be Unit Tested.